include/z64cutscene_commands.h

```diff
@@ -11,445 +11,263 @@
-    CMD_HH(messageId, startFrame), CMD_HH(endFrame, type), CMD_HH(topOptionBranch, bottomOptionBranch)
+#define CS_TEXT(textId, startFrame, endFrame, type, altTextId1, altTextId2) \
+    CMD_HH(textId, startFrame), CMD_HH(endFrame, type), CMD_HH(altTextId1, altTextId2)
 
-/**
- * ARGS
- *   s16 startFrame (s), s16 endFrame (e)
- * FORMAT
- *   FFFFssss eeeeFFFF FFFFFFFF
- *   size = 0xC
- */
 #define CS_TEXT_NONE(startFrame, endFrame) \
-    CS_TEXT_DISPLAY_TEXTBOX(0xFFFF, startFrame, endFrame, 0xFFFF, 0xFFFF, 0xFFFF)
+    CS_TEXT(CS_TEXT_ID_NONE, startFrame, endFrame, 0xFFFF, CS_TEXT_ID_NONE, CS_TEXT_ID_NONE)
 
-/**
- * ARGS
- *   s16 ocarinaSongAction (o), s16 startFrame (s), s16 endFrame (e), s16 topOptionBranch (i)
- * FORMAT
- *   oooossss eeee0002 iiiiFFFF
- *   size = 0xC
- */
 #define CS_TEXT_LEARN_SONG(ocarinaSongAction, startFrame, endFrame, messageId) \
-    CS_TEXT_DISPLAY_TEXTBOX(ocarinaSongAction, startFrame, endFrame, 0x0002, messageId, 0xFFFF)
+    CS_TEXT(ocarinaSongAction, startFrame, endFrame, CS_TEXT_OCARINA_ACTION, messageId, CS_TEXT_ID_NONE)
 
 /**
- * ARGS
- *   s16 transitionType (t), s16 startFrame (s), s16 endFrame (e)
- * FORMAT
- *   Capital U is Unused , endFrame duplicate
- *   0000002D 00000001 ttttssss eeeeUUUU
- *   size = 0x10
+ * Controls various types of screen transitions.
+ * @see `CutsceneTransitionTypes` for type options.
  */
-#define CS_SCENE_TRANS_FX(transitionType, startFrame, endFrame) \
-    CS_CMD_SCENE_TRANS_FX, 0x00000001, CMD_HH(transitionType, startFrame), CMD_HH(endFrame, endFrame)
+#define CS_TRANSITION(type, startFrame, endFrame) \
+    CS_CMD_TRANSITION, 0x00000001, CMD_HH(type, startFrame), CMD_HH(endFrame, endFrame)
 
 /**
- * ARGS
- *   s32 entries (e)
- * FORMAT
- *   00000056 eeeeeeee
- *   size = 0x8
+ * Declares a list of `CS_START_SEQ` entries.
  */
-#define CS_PLAY_BGM_LIST(entries) CS_CMD_PLAYBGM, CMD_W(entries)
+#define CS_START_SEQ_LIST(entries) \
+    CS_CMD_START_SEQ, CMD_W(entries)
 
 /**
- * ARGS
- *   s16 sequence (q), s16 startFrame (s), s16 endFrame (e)
- * FORMAT
- *   Capital U is Unused
- *   qqqqssss eeeeUUUU UUUUUUUU UUUUUUUU UUUUUUUU UUUUUUUU UUUUUUUU UUUUUUUU UUUUUUUU 00000000 00000000 00000000
- *   size = 0x30
+ * Starts a sequence at the specified time.
+ * @note The sequence ID is subtracted by 1 before being used. Add +1 to the desired sequence ID when passing it in.
  */
-#define CS_PLAY_BGM(sequence, startFrame, endFrame, unused0, unused1, unused2, unused3, unused4, unused5, unused6, unused7) \
-    CMD_HH(sequence, startFrame), CMD_HH(endFrame, unused0), \
-    CMD_W(unused1), CMD_W(unused2), CMD_W(unused3), CMD_W(unused4), CMD_W(unused5), \
-    CMD_W(unused6), CMD_W(unused7), 0x00000000, 0x00000000, 0x00000000
+#define CS_START_SEQ(seqIdPlusOne, frame, unused0, unused1, unused2, unused3, unused4, unused5, unused6, unused7, unused8) \
+    CMD_HH(seqIdPlusOne, frame), CMD_HH(unused0, unused1), \
```

Comment:
```suggestion
#define CS_START_SEQ(seqId, frame, unused0, unused1, unused2, unused3, unused4, unused5, unused6, unused7, unused8) \
    CMD_HH(seqId + 1, frame), CMD_HH(unused0, unused1), \
```
Is there a possibility of abstracting away this `plusOne` through the macros? Same would apply for `CS_STOP_SEQ`

Reply:
yea thats a good idea, but idk how to facilitate this change cause zapd currently has all of the + 1 values in scenes. not sure if theres some macro magic to do temporarily or what

Reply:
```c
#define CS_PLAY_BGM(sequence, startFrame, endFrame, unused0, unused1, \
unused2, unused3, unused4, unused5, unused6, unused7) \
CS_START_SEQ((sequence)-1, startFrame, endFrame, unused0, unused1, \
unused2, unused3, unused4, unused5, unused6, unused7)
```
:sunglasses: 
?